src/application/tester.go

```diff
@@ -26,45 +28,60 @@ func TestApp(appContext types.AppContext, logger *util.Logger, mode composebuild
 		if serviceContext.Config.Development.Scripts["test"] == "" {
 			logger.Logf("%s has no tests, skipping", serviceContext.Dir)
 		} else {
-			testPassed, err := TestService(serviceContext, logger, mode)
-			if err != nil {
+			testResult, err := TestService(serviceContext, logger, mode, shutdown)
+			switch {
```

Comment:
Thoughts on converting this to three ifs as I don't think these should be mutually exclusive. 

First, log the error if there is one. 
Second, return if interrupted (but don't return the error). 
Finally increment numFailed if needed